File: src/test/java/com/neuronrobotics/bowlerbuilder/GistUtilitiesTest.java
package com.neuronrobotics.bowlerbuilder;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

class GistUtilitiesTest {

  @Test
  void testIsValidCodeFileName1() {
    assertTrue(GistUtilities.isValidCodeFileName("a.b").isPresent());
  }

  @Test
  void testIsValidCodeFileName2() {
    assertFalse(GistUtilities.isValidCodeFileName("a").isPresent());
  }

  @Test
  void testIsValidCodeFileName3() {
    assertFalse(GistUtilities.isValidCodeFileName("a.").isPresent());
  }

  @Test
  void testIsValidCodeFileName4() {
    assertFalse(GistUtilities.isValidCodeFileName("").isPresent());
  }

}

File: src/test/java/com/neuronrobotics/bowlerbuilder/GistUtilitiesTest.kt
package com.neuronrobotics.bowlerbuilder

import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue

import org.junit.jupiter.api.Test

internal class GistUtilitiesTest {

    @Test
    fun testIsValidCodeFileName1() {
        assertTrue(GistUtilities.isValidCodeFileName("a.b").isPresent)
    }

    @Test
    fun testIsValidCodeFileName2() {
        assertFalse(GistUtilities.isValidCodeFileName("a").isPresent)
    }

    @Test
    fun testIsValidCodeFileName3() {
        assertFalse(GistUtilities.isValidCodeFileName("a.").isPresent)
    }

    @Test
    fun testIsValidCodeFileName4() {
        assertFalse(GistUtilities.isValidCodeFileName("").isPresent)
    }

}
